components/haptics/index.js

```diff
@@ -26,33 +26,23 @@ AFRAME.registerComponent('haptics', {
 
     this.callPulse = function () { self.pulse(); };
 
-    // There may exist a tracked-controls when this component is initialized
-    if (this.el.components['tracked-controls'] && this.el.components['tracked-controls'].controller) {
-      this.gamepad = this.el.components['tracked-controls'].controller;
-
-      if (this.gamepad.gamepad) {
+    var doInit = function (gamepad) {
+      self.gamepad = gamepad;
+      if (self.gamepad.gamepad) {
         // WebXR.
-        this.gamepad = this.gamepad.gamepad;
-      }
+         self.gamepad = self.gamepad.gamepad;
+       }
+       if (!self.gamepad || !self.gamepad.hapticActuators ||
+       !self.gamepad.hapticActuators.length) { return; }
+       self.addEventListeners();
+    };
 
-      if (!this.gamepad || !this.gamepad.hapticActuators ||
-          !this.gamepad.hapticActuators.length) { return; }
-      this.addEventListeners();
+    // There may exist a tracked-controls when this component is initialized
+    if (this.el.components['tracked-controls'] && this.el.components['tracked-controls'].controller) {
+      doInit(this.el.components['tracked-controls'].controller);
```

Comment:
Probably no need to pass the parameter and can be accessed directly within the function